app/assets/javascripts/match.js

```diff
@@ -0,0 +1,65 @@
+// Gets the GameId number from the URL and returns it.
+function getGameId() {
+  var fullUrl = document.URL;
+  var urlArray = fullUrl.split('/');
+  return urlArray[4];
+}
+
+function Match(id) {
+  var self = this;
+  window.onbeforeunload = function() { self.cancel() }
+  this.id = id;
+}
+
+Match.prototype.listenForConfirmation = function() {
+  var match = this;
+  setTimeout(function() { checkServer() }, 500);
+
+  function checkServer() {
+    console.log("Waiting for another player to join...");
+
+    $.ajax({
```

Comment:
This is not a good pattern, this double-ajax thing.  Check out `jQuery.when`.  It's an advanced feature of the `jQuery.Deferred` object.  It should allow you to stack two queries and only do some other callback _after_ they've both completed.  It's an implementation of a concept called a Promise.  Understand Promises in interviews for an extra $5,000/annum.